File: client/src/ts/packets/spawnentity.ts
import Packet, {PacketType} from './packet';

export class SpawnEntityPacket extends Packet {
    public x = 0.0;
    public y = 0.0
    public id = 0;
    public isSelf: boolean;

    constructor(packetDict: any) {
        super();

        this.x = packetDict['x'];
        this.y = packetDict['y'];
        this.id = packetDict['id'];
        this.isSelf = packetDict['isSelf'];
    }

    public dictify(): string {
        throw 5;
    }

    public getType(): PacketType {
        return PacketType.SpawnEntity;
    }
}

File: server/sockets/packets/spawn_entity.py
from server.game.player import Player
from server.sockets.packets import PacketType
from server.sockets.packets.packet import Packet


class SpawnEntityPacket(Packet):
    def __init__(self, id: int, is_self: bool, entity):
        self.id = id
        self.is_self = is_self
        self.entity = entity

        if isinstance(self.entity, Player):
            self.name = self.entity.name

    def dictify(self):
        values = dict(type=PacketType.SpawnEntity, id=self.id, isSelf=self.is_self, x=self.entity.x, y=self.entity.y)
        if self.name:
            values['name'] = self.name

        return values

    def get_type(self):
        return PacketType.SpawnEntity
